test/widgets/app_test.dart

```diff
@@ -37,24 +37,53 @@ void main() {
     }
 
     testWidgets('when no accounts, go to choose account', (tester) async {
+      check(testBinding.globalStore).accounts.isEmpty();
+      check(testBinding.globalStore).lastVisitedAccount.isNull();
       await prepare(tester);
       check(pushedRoutes).deepEquals(<Condition<Object?>>[
         (it) => it.isA<WidgetRoute>().page.isA<ChooseAccountPage>(),
       ]);
     });
 
-    testWidgets('when have accounts, go to home page for first account', (tester) async {
-      // We'll need per-account data for the account that a page will be opened
-      // for, but not for the other account.
```

Comment:
It's still the case that we don't need per-account data for the account that we're not opening a page for, right? So it'd be good to preserve this aspect, using `insertAccount` rather than `add`. That keeps the test a bit stronger: it would fail if ZulipApp accidentally started trying to load per-account data for all the accounts.